src/components/button/braintree.js

```diff
@@ -0,0 +1,34 @@
+/* @flow */
+
+import { SyncPromise } from 'sync-browser-mocks/src/promise';
+
+export type Braintree = {
+    client : {
+        create : Function
+    },
+    paypalCheckout : {
+        create : Function
+    }
+};
+
+export type BraintreePayPalClient = {
+    createPayment : Function,
+    tokenizePayment : Function
+};
+
+export function awaitBraintreeClient(braintree : Braintree, authorization : string) : SyncPromise<BraintreePayPalClient> {
+    return new SyncPromise((resolve, reject) => {
+
+        braintree.client.create({ authorization }, (err, client) => {
```

Comment:
Since v3.9.0, `braintree.client.create` and `braintree.paypalCheckout.create` will return a promise if you don't pass a callback, so this could just be:

```js
return braintree.client.create({ authorization })
  .then(client => braintree.paypalCheckout.create({ client }));
```